lib/LowerAddrSpaceCastPass.cpp

```diff
@@ -350,20 +350,92 @@ Value *clspv::LowerAddrSpaceCastPass::visitPtrToIntInst(PtrToIntInst &I) {
   return ptrToInt;
 }
 
+static bool DependsOnPhiNode(const Value *Value, const PHINode *PhiNode,
+                             DenseSet<const llvm::Value *> &Visited) {
+  if (!isa<Instruction>(Value)) {
+    return false;
+  }
+  if (Visited.contains(Value)) {
+    return false;
+  }
+
+  Visited.insert(Value);
+
+  // If the value is the PHI node itself, return true
+  if (Value == PhiNode) {
+    return true;
+  }
+
+  // Recursively check if any of the operands depend on the PHI node
+  for (const auto *Op : cast<Instruction>(Value)->operand_values()) {
+    if (DependsOnPhiNode(Op, PhiNode, Visited)) {
+      return true;
+    }
+  }
```

Comment:
I would prefer to avoid recursion when possible

Reply:
Replaced